src/sentry/api/endpoints/ai_unit_test_generation.py

```diff
@@ -0,0 +1,138 @@
+        }
+    }
+
+    def _respond_with_error(self, reason: str, status: int):
+        return Response(
+            {
+                "detail": reason,
+            },
+            status=status,
+        )
+
+    def _call_unit_test_generation(
+        self,
+        user: User | AnonymousUser,
+        owner: str | None,
+        name: str | None,
+        external_id: str | None,
+        pr_id: int,
+    ):
+        path = "/v1/automation/codegen/unit-tests"
+        body = orjson.dumps(
+            {
+                "repo": {
+                    "provider": "github",
+                    "owner": owner,
+                    "name": name,
+                    "external_id": external_id,
+                },
+                "pr_id": pr_id,
+                "invoking_user": (
+                    {
+                        "id": user.id,
+                        "display_name": user.get_display_name(),
+                    }
+                    if not isinstance(user, AnonymousUser)
+                    else None
+                ),
+            },
+            option=orjson.OPT_NON_STR_KEYS,
+        )
+
+        url, salt = get_seer_salted_url(f"{settings.SEER_AUTOFIX_URL}{path}")
+        response = requests.post(
+            url,
+            data=body,
+            headers={
+                "content-type": "application/json;charset=utf-8",
+                **sign_with_seer_secret(
+                    salt,
+                    body=body,
+                ),
+            },
+        )
+
+        response.raise_for_status()
+
+        return response.json().get("run_id")
+
+    def post(
+        self,
+        request: Request,
+        *args,
+        **kwargs,
+    ) -> Response:
+        owner = kwargs.get("github_org")
+        repo_name = kwargs.get("repo_name")
+        pull_request_number = kwargs.get("pull_request_number")
+        external_id = kwargs.get("external_id")
+        created_at = datetime.now().isoformat()
+
+        if pull_request_number is None:
+            return self._respond_with_error("Missing pull request number.", 400)
+
+        try:
+            pr_id = int(pull_request_number)
+        except ValueError:
+            return self._respond_with_error("Invalid pull request number.", 400)
+
+        try:
+            self._call_unit_test_generation(
+                request.user,
+                owner=owner,
+                name=repo_name,
+                external_id=external_id,
+                pr_id=pr_id,
+            )
+        except Exception as e:
+            logger.exception(
+                "Failed to send test generation request to seer",
+                extra={
+                    "created_at": created_at,
+                    "exception": e,
+                },
+            )
+            return self._respond_with_error(
+                "Test generation failed to start.",
+                500,
+            )
```

Comment:
you shouldn't need to manually do any of this. errors are already handled and 500s logged at the middleware level https://github.com/getsentry/sentry/blob/07396b7608ee6a7857a35641626b70027f6bc95d/src/sentry/api/base.py#L324